src/openai/azure/_async_client.py

```diff
@@ -353,8 +353,14 @@ async def create(
             extra_body=extra_body,
             timeout=timeout
         )
+        if isinstance(response, AsyncStream):
+            response._cast_to = AzureChatCompletionChunk  # or rebuild the stream?
```

Comment:
Yeah, this is where we want to add a compat clause/understand what the actual public contract of the streaming API is. If it is only intended to be an async iterable, then a yield for each item would have been enough. But since it is typed as a Stream, presumably the intent is also that the `response` attribute is available.

Reply:
...in the meantime, we can look at overriding `client._process_response_data`.

Reply:
Agreed on understanding the contract. it works as-is now, but I've definitely upped the amount of _internal things I've been touching...